sync/src/synchronizer/block_fetcher.rs

```diff
@@ -34,6 +34,16 @@ impl BlockFetcher {
     }
 
     pub fn peer_best_known_header(&self) -> Option<HeaderView> {
+        // update best known header and clean up unknown list
```

Comment:
I agree it. We also can skip checking the unknown list when the node is not in IBD.

Reply:
It seems inefficient that we take the whole list out, and then insert back one by one. We can peek the unknown hash backward. If there's a hash found in the local index, we pop it and add it as a new received block. Otherwise the loop exists early.

```
while (!peer.unknown_list.empty()) {
  if index.exists(peer.unknown_list[-1]) {
    receive_new_block(peer.unknown_list.pop());
  } else {
    return;
  }
}
```